### DISCUSSION TITLE
(Windows) Use Standard Directory values in ExeCommand?

### DISCUSSION THREAD

**Discussion by Zerowalker** [wixtoolset/issues#8170] [Questions]
I Got something like this:

<?define name = "...."?>
<?define exe = "...."?>
<?define path = "...."?>
<?define configPath = "...."?>
<?define silentArgs = ""?>
<?define version = "0.1.2"?>
<?define publisher = "...."?>
<?define guid = "...."?>

<Wix xmlns="http://wixtoolset.org/schemas/v4/wxs">

        <Package Name="$(var.name)" Manufacturer="$(var.publisher)" Version="$(var.version)"
                UpgradeCode="$(var.guid)" Scope="perMachine">
                <Media Id="1" Cabinet="setup.cab" EmbedCab="yes" />

                <Feature Id="InstallFeature" Title="Install Feature">
                        <ComponentRef Id="file" />
                        <ComponentRef Id="file2" />
                </Feature>

                <CustomAction Id="run_exe" Return="asyncWait" Execute="commit"
                        ExeCommand='create $(var.name) binpath="$(var.path)\$(var.exe)" start=auto' HideTarget="yes" Impersonate="no"
                        BinaryRef="sc" />
                <CustomAction Id="uninstall_exe" Return="asyncWait" Execute="commit"
                        ExeCommand='delete $(var.name)' HideTarget="yes" Impersonate="no"
                        BinaryRef="sc" />
                <InstallExecuteSequence>
                        <Custom Action="run_exe" After="InstallFiles"
                                Condition="NOT Installed AND NOT REMOVE" />
                        <Custom Action="uninstall_exe" After="InstallFiles"
                                Condition="Installed AND REMOVE" />
                </InstallExecuteSequence>

                <Binary Id="sc" SourceFile="C:\Windows\System32\sc.exe" />
                
                <StandardDirectory Id="ProgramFiles64Folder">
                        <Directory Id="INSTALLLOCATION" Name="$(var.name)">
                                <Component Id="file" DiskId="1">
                                    <File Id="executable" Name="$(var.exe)" Source="$(var.path)\$(var.exe)" />
                                </Component>
                                 <Component Id="file2" DiskId="1">
                                    <File Id="config" Name="config.toml" Source="$(var.configPath)\config.toml"/>
                                </Component>
                        </Directory>
                </StandardDirectory>
        </Package>
</Wix>

The files are installed into "programfiles\$(var.name)".
And i want to create a service to the exe in that path but I don't know how to point there without manually setting the whole path.
(see "ExeCommand" in "CustomAction" run_exe)

It would also be nice to use a Standard Directory for the "Binary" so I can use "SystemFolder" instead of manually setting "C:\Windows\System32".

I might just be going at this the wrong way of course, so please do tell if that's the case, I'm a complete beginner at this:)

Thanks!

**Comment by jcasale** [comment]
Why are you using a CA with sc.exe to manage services?

See https://wixtoolset.org/docs/schema/wxs/serviceinstall/.
